parcels/kernel.py

```diff
@@ -226,7 +226,7 @@ def execute_python(self, pset, endtime, dt):
                 try:
                     p.dt = sign_dt * dt_pos
                     res = self.pyfunc(p, pset.fieldset, p.time)
-                except FieldSamplingError as fse:
+                except FieldOutOfBoundError as fse:
```

Comment:
Also catch `FieldSamplingError` here?

Reply:
It's not needed (at least for now).
FieldSamplingError should not really happen. If it happens, there is a specific message for FieldSamplingError.
If user wants to treat this error, it can do it with the Error recovery kernel (not really used so far, so no need to add a specific one for this error)